integration/lifecycle_test.go

```diff
@@ -312,6 +312,140 @@ var _ = Describe("Lifecycle", func() {
+			},
+			ConfigModifier: func(c *container.Config) {
+				c.Tty = true
+			},
+			Env: map[string]string{
+				"CNB_LAYERS_DIR": "/home/ubuntu/layers",
+				"CNB_APP_DIR":    "/home/ubuntu/workspace2",
+				"CNB_STACK_ID":   "cflinuxfs4",
+				"CNB_USER_ID":    "1000",
+				"CNB_GROUP_ID":   "1000",
+				"CNB_LOG_LEVEL":  "DEBUG",
+				"CNB_NO_COLOR":   "true",
+			},
+			LifecycleHooks: []testcontainers.ContainerLifecycleHooks{
+				{
+					PostStarts: []testcontainers.ContainerHook{
+						func(ctx context.Context, container testcontainers.Container) error {
+							code, _, err := container.Exec(ctx, []string{"apt", "update"})
+							if code != 0 {
+								return fmt.Errorf("failed to run apt update, RC: %d", code)
+							}
+
+							return err
+						},
+						func(ctx context.Context, container testcontainers.Container) error {
+							code, _, err := container.Exec(ctx, []string{"apt", "install", "ca-certificates", "skopeo", "-y"})
+							if code != 0 {
+								return fmt.Errorf("failed to run install ca-certificates, RC: %d", code)
+							}
+
+							return err
+						},
+
+						func(ctx context.Context, container testcontainers.Container) error {
+							if err := container.CopyFileToContainer(ctx, "../bin/builder", "/tmp/builder", 0o755); err != nil {
+								return err
+							}
+
+							if err := container.CopyFileToContainer(ctx, "../bin/launcher", "/tmp/launcher", 0o755); err != nil {
+								return err
+							}
+
+							if err := container.CopyDirToContainer(ctx, "./testdata/workspace2", "/home/ubuntu/", 0o755); err != nil {
+								return err
+							}
+
+							if err := runInContainer(ctx, container, "chown", "-R", "ubuntu:ubuntu", "/home/ubuntu/workspace2"); err != nil {
+								return err
+							}
+
+							if err := runInContainer(ctx, container, "mkdir", "-p", "/tmp/buildpacks"); err != nil {
+								return err
+							}
+
+							if err := runInContainer(ctx, container, "skopeo", "copy", "docker://gcr.io/paketo-buildpacks/go:latest", "oci:/tmp/buildpacks/10bfa3ba0b8af13e"); err != nil {
+								return err
+							}
+
+							if err := runInContainer(ctx, container, "chown", "-R", "ubuntu:ubuntu", "/tmp/buildpacks"); err != nil {
+								return err
+							}
+
+							return err
+						},
+					},
+				},
+			},
+		}
+		testContainer, err = testcontainers.GenericContainer(context.Background(), testcontainers.GenericContainerRequest{
+			ContainerRequest: req,
+			Started:          true,
+		})
+		Expect(err).To(BeNil())
+	})
+
+	It("web process is added even when buildpack doesn't add one", func() {
+		By("building the app", func() {
+			code, out, err := testContainer.Exec(context.Background(), []string{
+				"/tmp/builder",
+				"-b", "gcr.io/paketo-buildpacks/go",
+				"-r", "/tmp/build-result.json",
+				"-d", "/tmp/droplet.tgz",
+				"-l", "/home/ubuntu/layers",
+				"-w", "/home/ubuntu/workspace2",
+			}, exec.WithUser("ubuntu"))
+			Expect(err).To(BeNil())
+
+			if code != 0 {
```

Comment:
Nice. This is no show what error occured, right? Could we extract this to a `printErrorMsgIfExists(code, out)` (or something like this)? Probably it makes sense for every integration test.